jwst/photom/photom.py

```diff
@@ -1007,13 +1029,50 @@ def create_2d_conversion(self, model, exptype, conversion, waves, relresps,
         # 2D wavelength grid
         conv_2d = np.interp(wl_array, waves, relresps, left=np.nan, right=np.nan)
 
+        if include_dispersion:
+            dispaxis = get_dispersion_direction(self.exptype, self.grating, self.filter, self.pupil)
+            if dispaxis is not None:
+                dispersion_array = self.get_dispersion_array(wl_array, dispaxis)
+                # Convert dispersion from micron/pixel to angstrom/pixel
```

Comment:
This obviously makes the assumption that the delivered flux calibration data for NIRCam (and only NIRCam) grism modes is in units of "per Angstrom". Is there anyway to get meta info from the flux calibration reference file that confirms what units it's in? What happens if in the future the units of the calibration data change?

Reply:
The reference file doesn't have units for any of the columns.  I'm assuming that since the dispersion is known to change with wavelength, the team left out the dispersion term from the calculation of the zero point, assuming that the pipeline would calculate the dispersion as a function of wavelength and put the factor back in.  The fact that they report that the fluxes are wrong by a factor of ~10 means they are assuming the dispersion is in Angstroms/pixel, since the dispersion of the gratings is known to be about 10 Angstrom/pixel.  The team will now have to ensure that in later reference file deliveries, their reference files factor out the dispersion in Angstrom/pixel, as implemented in the pipeline here.